Lib/test/test_exceptions.py

```diff
@@ -1843,6 +1843,36 @@ def test_memory_error_in_subinterp(self):
         rc, _, err = script_helper.assert_python_ok("-c", code)
         self.assertIn(b'MemoryError', err)
 
+    @cpython_only
+    # Python built with Py_TRACE_REFS fail with a fatal error in
+    # _PyRefchain_Trace() on memory allocation error.
+    @unittest.skipIf(support.Py_TRACE_REFS, 'cannot test Py_TRACE_REFS build')
+    def test_exec_set_nomemory_hang(self):
+        import_module("_testcapi")
+        # gh-134163: Test case that triggers no memory hang condition
```

Comment:
Let's be a little more specific:
```suggestion
        # gh-134163: A MemoryError inside code that was wrapped by a try/except
        # block would lead to an infinite loop.
```